hs_core/views/utils.py

```diff
@@ -57,18 +58,22 @@ def authorize(request, res_id, edit=False, view=False, full=False, superuser=Fal
     except ObjectDoesNotExist:
         raise NotFound(detail="No resource was found for resource id:%s" % res_id)
 
-    if user.is_authenticated():
+    if discoverable and (edit is False and view is False and full is False and superuser is False):
+        authorized = res.raccess.discoverable
+    elif user.is_authenticated():
         has_edit = user.uaccess.can_change_resource(res)
         has_view = user.uaccess.can_view_resource(res)
```

Comment:
@selimnairb Yes, this is in line with what I am thinking. And I agree it is a good thing we use this authorize() function as the single place to check for authorization. I am looking at the logic you added regarding discoverable in authorize() again more carefully, and what you added with discoverable parameter check makes sense to me now.